scripts/halium

```diff
@@ -514,49 +547,57 @@ mountroot() {
 		tell_kmsg "Normal boot"
 
 		mount --move /halium-system ${rootmnt}
-
-		# Mount some tmpfs
 		mkdir -p ${rootmnt}/android
-		mount -o rw,size=4096 -t tmpfs none ${rootmnt}/android
-
-		# Create some needed paths on tmpfs
-		mkdir -p ${rootmnt}/android/data ${rootmnt}/android/system
 
 		# Mounting userdata outside of /android, to avoid having LXC container access it
 		mkdir -p ${rootmnt}/userdata
 		mount --move /tmpmnt ${rootmnt}/userdata
 
+		if [ $ANDROID_IMAGE_MODE = "system" ]; then
+			# Mount some tmpfs
+			mount -o rw,size=4096 -t tmpfs none ${rootmnt}/android
+			# Create some needed paths on tmpfs
+			mkdir -p ${rootmnt}/android/data ${rootmnt}/android/system
+			device=$(extract_device_name /android-system)
+		elif [ $ANDROID_IMAGE_MODE = "rootfs" ]; then
+			# In rootfs mode system.img is / of android, so move it where it should be
+			mount --move /android-rootfs ${rootmnt}/android
+			device=$(extract_device_name ${rootmnt}/android/system)
+		fi
+		tell_kmsg "device is $device"
+
 		# Create a fake android data, shared by rootfs and LXC container
 		mkdir -p ${rootmnt}/userdata/android-data
 		mount -o bind ${rootmnt}/userdata/android-data ${rootmnt}/android/data
-		[ ! -h ${rootmnt}/data ] && ln -sf /android/data ${rootmnt}/data
+		[ ! -h ${rootmnt}/data ] && ln -sf ${rootmnt}/android/data ${rootmnt}/data
```

Comment:
This symbolic link will be invalid after the chroot: are you sure about this modification?